src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -166,37 +175,67 @@ export class EditorComponent implements OnInit {
       this.scaleFactor =  1;
     }
 
-    //  Canvas to draw mask, hidden.
+    // Canvas to draw mask, hidden.
     this.maskCanvas.nativeElement.width = imgWidth;
     this.maskCanvas.nativeElement.height = imgHeight;
     this.maskCtx = this.maskCanvas.nativeElement.getContext('2d');
     this.maskCtx.lineCap = this.maskCtx.lineJoin = 'round';
     this.maskCtx.strokeStyle = this.MAGENTA;
 
-    //  Canvas to show mask scaled.
+    // Canvas to show mask scaled.
     this.scaledCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.scaledCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.scaledCtx = this.scaledCanvas.nativeElement.getContext('2d');
 
-    //  Canvas to show Image (never changes unless user only wants to see Mask)
+    // Canvas to show Image (never changes unless user only wants to see Mask)
     this.imageCanvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.imageCanvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.imageCtx = this.imageCanvas.nativeElement.getContext('2d');
-    
+
+    // Canvas to paint cursor-overlay of brush size.
+    this.cursorCanvas.nativeElement.width = imgWidth * this.scaleFactor;
+    this.cursorCanvas.nativeElement.height = imgHeight * this.scaleFactor;
+    this.cursorCtx = this.cursorCanvas.nativeElement.getContext('2d');
+
     this.stageWidth = imgWidth * this.scaleFactor;
     this.stageHeight = imgHeight * this.scaleFactor;
 
-    //   Draws image non scaled on full canvas
+    // Draws image non scaled on full canvas
     this.imageCtx.drawImage(this.image, 0, 0);
 
-    //  Only gets the image data from (0,0) to (width,height) of image.
+    // Only gets the image data from (0,0) to (width,height) of image.
     this.originalImageData = this.imageCtx.getImageData(0, 0, imgWidth, imgHeight);
 
     this.drawScaledImage();
     console.log('put imagedata');
   }
+
+
+  /* The following 2 functions: Handles cursor tracking and resizing. */
   
- /**
+  // Draws/Redraws 'cursor' at the current position of user's mouse.
+  setCursorPosition(e: MouseEvent): void {
+    // These are the coordinates used to paint.
+    this.cursorX = e.offsetX;
+    this.cursorY = e.offsetY;
+
+    this.cursorCtx.clearRect(0, 0, this.stageWidth, this.stageHeight);
+
+    this.cursorCtx.beginPath();
+    this.cursorCtx.arc(this.cursorX, this.cursorY,
+        this.brushWidth * this.scaleFactor * .5, 0, 2 * Math.PI, true);
```

Comment:
Does the cursor become invisible if the radius is too small? I wonder if we want to clamp it to be at least 1.

Reply:
The cursor is still visible at the smallest brush size, which would be a value of 1.

Reply:
But what if scaleFactor is less than 1? I can definitely make the cursor disappear on the live version of the app.